src/shell/shell_cmds.cpp

```diff
@@ -240,16 +238,36 @@ void DOS_Shell::CMD_HELP(char * args){
 	HELP("HELP");
 	bool optall=ScanCMDBool(args,"ALL");
 	/* Print the help */
-	if (!optall) WriteOut(MSG_Get("SHELL_CMD_HELP"));
+	if (!optall && !*args) WriteOut(MSG_Get("SHELL_CMD_HELP"));
+	upcase(args);
 	Bit32u cmd_index=0,write_count=0;
+	bool show = false;
 	while (cmd_list[cmd_index].name) {
-		if (optall || !cmd_list[cmd_index].flags) {
-			WriteOut("<\033[34;1m%-8s\033[0m> %s",cmd_list[cmd_index].name,MSG_Get(cmd_list[cmd_index].help));
-			if (!(++write_count % 24))
-				CMD_PAUSE(empty_string);
+		if (optall || (*args && !strcmp(args, cmd_list[cmd_index].name)) || (!*args && !cmd_list[cmd_index].flags)) {
+			if (*args && !strcmp(args, cmd_list[cmd_index].name) && !optall) {
+				std::string cmd = args;
+				if (cmd == "CD") cmd = "CHDIR";
+				else if (cmd == "DEL" || cmd == "ERASE") cmd = "DELETE";
+				else if (cmd == "LH") cmd = "LOADHIGH";
+				else if (cmd == "MD") cmd = "MKDIR";
+				else if (cmd == "RD") cmd = "RMDIR";
+				else if (cmd == "REN") cmd = "RENAME";
+				WriteOut("%s\n",MSG_Get(cmd_list[cmd_index].help));
+				const char* long_m = MSG_Get(("SHELL_CMD_" + cmd + "_HELP_LONG").c_str());
+				if (strcmp("Message not Found!\n",long_m)) WriteOut(long_m);
+				else WriteOut("%s\n", cmd.c_str());
+				show = true;
+				break;
+			} else {
+				WriteOut("<\033[34;1m%-8s\033[0m> %s",cmd_list[cmd_index].name,MSG_Get(cmd_list[cmd_index].help));
+				if (!(++write_count % 24))
+					CMD_PAUSE(empty_string);
+			}
```

Comment:
This should be formatted using `./scripts/format-commit.sh` (if you've got a *nix system), or VS2019 should find the `.clang_format` file and offer to format.

Here's a drop-in reformatted replacement, if either above options aren't available:

``` c++
void DOS_Shell::CMD_HELP(char * args){
	HELP("HELP");
	bool optall = ScanCMDBool(args, "ALL");
	/* Print the help */
	if (!optall && !*args)
		WriteOut(MSG_Get("SHELL_CMD_HELP"));
	upcase(args);
	Bit32u cmd_index = 0, write_count = 0;
	bool show = false;
	while (cmd_list[cmd_index].name) {
		if (optall || (*args && !strcmp(args, cmd_list[cmd_index].name)) ||
		    (!*args && !cmd_list[cmd_index].flags)) {
			if (*args && !strcmp(args, cmd_list[cmd_index].name) &&
			    !optall) {
				std::string cmd = args;
				if (cmd == "CD")
					cmd = "CHDIR";
				else if (cmd == "DEL" || cmd == "ERASE")
					cmd = "DELETE";
				else if (cmd == "LH")
					cmd = "LOADHIGH";
				else if (cmd == "MD")
					cmd = "MKDIR";
				else if (cmd == "RD")
					cmd = "RMDIR";
				else if (cmd == "REN")
					cmd = "RENAME";
				WriteOut("%s\n", MSG_Get(cmd_list[cmd_index].help));
				const char *long_m = MSG_Get(
				        ("SHELL_CMD_" + cmd + "_HELP_LONG").c_str());
				if (strcmp("Message not Found!\n", long_m))
					WriteOut(long_m);
				else
					WriteOut("%s\n", cmd.c_str());
				show = true;
				break;
			} else {
				WriteOut("<\033[34;1m%-8s\033[0m> %s",
				         cmd_list[cmd_index].name,
				         MSG_Get(cmd_list[cmd_index].help));
				if (!(++write_count % 24))
					CMD_PAUSE(empty_string);
			}
		}
		cmd_index++;
	}
	char p[2] = {0};
	if (!optall && *args && !show)
		CMD_HELP(p);
}
```

Reply:
I'm having a hard time easily following all the conditional logic - maybe each chunk could be assigned to a `bool` variable that has a name describing the condition? 

For example,
What kind of command line has a user typed to produce this:
`optall || (*args && !strcmp(args, cmd_list[cmd_index].name)`
versus this:
`(!*args && !cmd_list[cmd_index].flags))`
?

Reply:
Is it possible to address both of these conditions:

``` c++
	if (!optall && !*args)
		WriteOut(MSG_Get("SHELL_CMD_HELP"));
```

and
``` c++
	if (!optall && *args && !show)
		CMD_HELP(p);
```

Early on in the function, and return immediately after processing them?

That mean we can eliminate those conditions in the while loop (maybe simplifying it).

Reply:
> This should be formatted using ./scripts/format-commit.sh (if you've got a *nix system), or VS2019 should find the .clang_format file and offer to format.

Thanks a lot for the replacement! I skipped the format-commit.sh for this part as other parts were already formatted and I focused on the function of this, but definitely should format them all.

> I'm having a hard time easily following all the conditional logic - maybe each chunk could be assigned to a `bool` variable that has a name describing the condition?
> 
> For example, what kind of command would a user have to type to produce this condition:
> `optall || (*args && !strcmp(args, cmd_list[cmd_index].name)`
> 
> ... versus this condition:
> `(!*args && !cmd_list[cmd_index].flags))`

I think it is definitely a good idea to assign them bool variables! Done.

> Is it possible to address both of these conditions:
> 
> ```c++
> 	if (!optall && !*args)
> 		WriteOut(MSG_Get("SHELL_CMD_HELP"));
> ```
> 
> and
> 
> ```c++
> 	if (!optall && *args && !show)
> 		CMD_HELP(p);
> ```
> 
> Early on in the function, and return immediately after processing them?
> 
> That means we can eliminate those conditions in the while loop (maybe simplifying it).

I think the first one is already before the whole loop, whereas the second one cannot be returned immediately unless there is a separate loop for it. So perhaps it is already a simple solution.